okhttp/src/main/kotlin/okhttp3/internal/platform/ConscryptPlatform.kt

```diff
@@ -50,7 +50,7 @@ class ConscryptPlatform private constructor() : Platform() {
       "Unexpected default trust managers: ${trustManagers.contentToString()}"
     }
     val x509TrustManager = trustManagers[0] as X509TrustManager
-    Conscrypt.setHostnameVerifier(x509TrustManager) { _, _ -> true }
+    Conscrypt.setHostnameVerifier(x509TrustManager) { _, _, _ -> true }
```

Comment:
I think you can make this work with both versions if you’re careful. Implement an interface in a class that defines both the old and the new method, and apply `override` on only one of the two methods. At runtime things should just work.